### ISSUE TITLE
Improve documentation for twisted.protocols.htb

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5655]
|<img alt="gsk's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| gsk reported|
|-|-|
|Trac ID|trac#5655|
|Type|enhancement|
|Created|2012-04-25 00:45:37Z|
|Branch|https://github.com/twisted/twisted/tree/htb-docs-5655|

Hello,
I was using the htb module and figured the documentation could use an update. I updated the docstrings in the following classes: Bucket, HierarchicalBucketFilter, IBucketFilter, FilterByHost and FilterByServer. Specifically, I got rid of all mentions of "bytes" and replaced them with "tokens" and removed all "I" pronouns. There are also a few lines of code that were refactored to be more easily understandable.

Attachments:

* [bucket_docs.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3f4/3f4894cace26b4947441e83a316d9cbd9443c340/f4e83b945ec81b5ba4d8b3d22b28fec5fde3e4c0.patch) (4850 bytes) - added by gsk on 2012-04-25 00:46:31Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5655 5655
type__enhancement enhancement
reporter__gsk gsk
priority__normal normal
milestone__ 
branch__branches_htb_docs_5655 branches/htb-docs-5655
branch_author__gsk__thijs gsk, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1335314737000000 1335314737000000
changetime__1342457531000000 1342457531000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [34320]) Branching to 'htb-docs-5655'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [34321]) apply bucket_docs.patch, adjust module to coding standard. refs [#5655](https://github.com/twisted/twisted/issues/5655)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [34322]) add news file, refs [#5655](https://github.com/twisted/twisted/issues/5655)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Thanks for your patch. I made some additional coding standard changes and putting it up for review again.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @thijstriemstra|
|-|-|

1.
```
+    I filter things into buckets, and I am nestable

+    I wrap a Transport and shape the rate at which it receives data.

+    I dispense Protocols with traffic shaping on their transports.
```

Please clean Those docstrings as well, removing the "I" usage.

2.

```
+            delta_time = now - self.lastDrip
+            delta_tokens = delta_time * self.rate
+            self.content = max(0, self.content - delta_tokens)
```

The variables names should be deltaTime and deltaTokens.

3.
```
+    Implementation of a Token Bucket.

+    @cvar maxburst: The maximum number of tokens that the Bucket can
+        hold at any given time. If this is None, the Bucket has

+    @cvar rate: The rate at which the Bucket drains, in number
+        of tokens per second. If the rate is None, the Bucket
```

Using a `Bucket` with capital is unnecessary. You should simply use `bucket` there.

It looks good, please make those small changes, have a buildbot run of the API doc and merge if green. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [34458]) address review comments, refs [#5655](https://github.com/twisted/twisted/issues/5655)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra removed owner|
|-|-|

Addressed the review comments and made some additional fixes, putting them up for review again. Also tried to run a build but the [doc builder](http://buildbot.twistedmatrix.com/builders/documentation) is offline..

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for reporting and submitting a patch for this, gsk, and thanks thijs for finishing it up. A few comments:

 * There was one pydoctor issue:

```
twisted.protocols.htb.Bucket.add:65 invalid ref to amount
```

 * Can you mark up `rate` in:

```
+        The L{Bucket} drains at the rate specified by the class
+        variable rate.
```

 * Can you mark up `Bucket` in:

```
+        Find or create a Bucket corresponding to the provided parameters.
```

 * Since Consumer, Transport, and Protocols are capitalized it implies that you mean something specific. Can you either add C{} markup for them or L{} markup to their interface classes?

```
+    Wraps a Consumer and shapes the rate at which it receives data.

+    Wraps a Transport and shapes the rate at which it receives data.

+    Dispense Protocols with traffic shaping on their transports.
```

After those changes, I think this is ready to merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [34854]) address review comments, refs [#5655](https://github.com/twisted/twisted/issues/5655)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Thanks. [Build results](http://buildbot.twistedmatrix.com/builders/documentation/builds/2269) look good, merging.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [34855]) Merge htb-docs-5655: Updated the API documentation for twisted.protocols.htb.

Author: gsk, thijs
Reviewer: jesstess
Fixes: [#5655](https://github.com/twisted/twisted/issues/5655)
